device/globals/player.gd

```diff
@@ -345,12 +345,14 @@ func _ready():
 	if Engine.is_editor_hint():
 		return
 
-	animation = get_node("animation")
 	vm = $"/root/vm"
-	vm.register_object("player", self)
-	#_update_terrain();
+
 	if has_node("animation"):
-		get_node("animation").connect("animation_finished", self, "anim_finished")
+		animation = $"animation"
+		animation.connect("animation_finished", self, "anim_finished")
+		assert(animations)
```

Comment:
This is a good improvement. Might be better to use `vm.report_errors()` though, to provide the developer with a more helpful error message?